packages/@react-spectrum/textfield/src/TextArea.tsx

```diff
@@ -28,22 +28,22 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
     onChange,
     ...otherProps
   } = props;
-
   // not in stately because this is so we know when to re-measure, which is a spectrum design
   let [inputValue, setInputValue] = useControlledState(props.value, props.defaultValue, () => {});
 
+  let inputText = props.value ?? props.defaultValue ?? ''; 
   let inputRef = useRef<HTMLTextAreaElement>();
 
   let onHeightChange = useCallback(() => {
-    if (isQuiet) {
+    if (isQuiet || isReadOnly) {
```

Comment:
So when the conditional is change to `|| !isReadOnly`, then one of the test cases fails, whereas if it's kept `|| isReadOnly`, it passes. Not sure exactly why this is happening, I agree that `|| isReadOnly` isn't exactly representing what we're trying to solve. 

TextArea › default does not change height
    expect(received).toBe(expected) // Object.is equality
    Expected: ""
    Received: "500px"

^this is what is printed for the failing test case

Reply:
I think maybe the conditional should be tweaked a bit more then?

Reply:
I don't think that's quite right, if `isQuiet` is true, it returns true before the `or`
if `isQuiet` is false, then we evaluate the right side of the `or`, but it's false because it's an `and` with a false `isQuiet`

I revisited it, I'm pretty sure it should actually be `if (isQuiet && !isReadOnly) {`
Reasons being, when it's readonly we're going do our calculations down in Field's ReadOnlyField. So when that is happening, we don't want to run this height calculation in TextArea. We do want to run it when the field is quiet though. So in English it's just we only want to calculate the height here when the component isQuiet and the readonly field isn't doing this calculation for us.

So boolean logic table would be (comparing AND vs OR original solution)
```
AND
x y` | xy
--------
0 0  | 0
1 0  | 1
0 1  | 0
1 1  | 0

OR
x y | x + y
--------
0 0  | 0
1 0  | 1
0 1  | 1
1 1  | 1
```
So we can see that the OR and the AND were not equivalent, just in case you were curious. You can also do this for the proposal you had in your last comment.